sktime/forecasting/squaring_residuals.py

```diff
@@ -0,0 +1,399 @@
+    the minimal number of observations to which the forecaster is fitted.
+
+    1. For :math:`i = initial\_window, \dots, N - steps\_ahead`
+        a. Train/Update forecaster A on :math:`y(t_1), \dots, y(t_i)`
+        b. Make point prediction for :math:`t_{i+steps\_ahead}` to get
+           :math:`\hat{y}(t_{i+steps\_ahead})`
+        c. Compute the residual for :math:`t_{i+steps\_ahead}` with
+           :math:`r(t_{i+steps\_ahead}) := y(t_{i+steps\_ahead})
+           - \hat{y}(t_{i+steps\_ahead})`
+        d. Apply  :math:`e(t_{i+steps\_ahead}) := h(r(t_{i+steps\_ahead}))`
+           where :math:`h(x)` is given by :math:`strategy`
+    2. Train `variance_forecaster` on
+       :math:`e(t_{initial\_window+steps\_ahead}), \dots, e(t_{N})`
+
+    Prediction for :math:`t_{N+steps\_ahead}` is done as follows:
+
+    1. Use `forecaster` to predict location param :math:`\hat{y}(t_{N+steps\_ahead})`
+    2. Use `variance_forecaster` to predict scale param :math:`e(t_{N+steps\_ahead})`
+    3. Calculate prediction intervals based on e.g. normal assumption
+       :math:`N(\hat{y}(t_{N+steps\_ahead}),  e(t_{N+steps\_ahead}))`
+
+    Parameters
+    ----------
+    forecaster : sktime.estimator, optional
+        Estimator to which probabilistic forecasts are being added
+    variance_forecaster : sktime.estimator, optional
+        Estimator which is fitted to the residuals of forecaster
+    initial_window : int, optional, default=2
+        Size of initial_window to which forecaster is fitted
+    steps_ahead : int, optional, default=1
+        Steps ahead for which we predict the residuals
+    strategy : str, optional, default='square'
+        Function applied to the residuals
+    distr : str, optional, default='norm'
+        Distributional assumption (["norm", "laplace", "t", "cauchy"])
+    distr_kwargs : dict, optional
+        Additional arguments required by the distributional assumption
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_macroeconomic
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.theta import ThetaForecaster
+    >>> from sktime.forecasting.squaring_residuals import SquaringResiduals
+    >>> fc = NaiveForecaster()
+    >>> var_fc = ThetaForecaster()
+    >>> y = load_macroeconomic().realgdp
+    >>> sqr = SquaringResiduals(forecaster=fc, variance_forecaster=var_fc)
+    >>> sqr = sqr.fit(y)
+    >>> fh = ForecastingHorizon(values=[1, 2, 3])
+    >>> pred_interval = sqr.predict_interval(fh, coverage=0.95)
+    """
+
+    _tags = {
+        "scitype:y": "univariate",  # which y are fine? univariate/multivariate/both
+        "ignores-exogeneous-X": True,  # does estimator ignore the exogeneous X?
+        "handles-missing-data": False,  # can estimator handle missing data?
+        "y_inner_mtype": "pd.Series",  # which types do _fit, _predict, assume for y?
```

Comment:
why can´t we do this multivariate? Should not be too hard, use `pd.DataFrame` here.
(maybe I don´t see a problem though)